### PR TITLE
GUVNOR-3003: [Library] [Assets list] Order assets alphabetically

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/GUVNOR-3003

### CODE REVIEW

**Comment by manstis** [comment]
Part of an ensemble:

- https://github.com/AppFormer/uberfire/pull/697
- https://github.com/kiegroup/kie-wb-common/pull/866
- https://github.com/kiegroup/drools-wb/pull/493
- https://github.com/kiegroup/jbpm-designer/pull/595
- https://github.com/kiegroup/jbpm-form-modeler/pull/105
- https://github.com/kiegroup/optaplanner-wb/pull/163

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by ederign** [comment]
Jenkins please retest this.

**Comment by manstis** [comment]
@ederign None of the ```kiegroup``` PRs are likely to build until https://github.com/AppFormer/uberfire/pull/697 is merged (IIRC Jenkins doesn't cross the ```kiegroup```\```appformer``` boundary when building related PRs).

**Comment by ederign** [comment]
@manstis I'm almost sure that jenkins can cross the boundaries if you use the same branch name (like you did). My guess is because we still have our old friend test failing.

@manstis can I merge https://github.com/AppFormer/uberfire/pull/697 in isolation?

**Comment by manstis** [comment]
@ederign Yeah.. so I see. I submitted what I _thought_ was going to be a fix, but looks like it isn't :-(

AppFormer's PR can however be merged in isolation of these without problem.

**Comment by ederign** [comment]
jenkins retest this

**Comment by manstis** [comment]
@ederign OK, looks like I have 2 broken tests... I'll take a look tomorrow.

File: kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/main/java/org/kie/workbench/common/services/refactoring/backend/server/indexing/DefaultIndexBuilder.java [hunk 1]
```diff
@@ -55,43 +56,120 @@ public DefaultIndexBuilder addGenerator(final IndexElementsGenerator generator)
...
+                        return true;
+                    }
+                });
             }
         }
 
         if (pkgName == null) {
             pkgName = pkg.getPackageName();
         }
         if (pkg != null) {
```
Comment (jomarko) [inline kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/main/java/org/kie/workbench/common/services/refactoring/backend/server/indexing/DefaultIndexBuilder.java]: Hi @manstis shouldn't be here `pkgName != null` ?

File: kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/IndexingTest.java [hunk 2]
```diff
@@ -181,6 +179,35 @@ protected IOService ioService() {
...
+//            final TestIndexer indexer = getIndexer();
+//            IndexersFactory.clear();
+//            IndexersFactory.addIndexer(indexer);
+//
+//            //Mock CDI injection and setup
+//            indexer.setIOService(ioService);
+//            indexer.setProjectService(getProjectService());
+//            indexer.setResourceTypeDefinition(getResourceTypeDefinition());
+//        }
+//        return ioService;
```
Comment (jomarko) [inline kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/IndexingTest.java]: Do we want to preserve this long commented code?

File: kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/resources/org/kie/workbench/common/services/refactoring/backend/server/query/findresources/DRL4.drl [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package org.kie.workbench.mock.package;
```
Comment (jomarko) [inline kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/resources/org/kie/workbench/common/services/refactoring/backend/server/query/findresources/DRL4.drl]: It seems to me that this asset is not needed. Am I missing something?

File: kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/resources/org/kie/workbench/common/services/refactoring/backend/server/query/findresources/RULE4.drl [hunk 4]
```diff
@@ -0,0 +1,28 @@
+package org.kie.workbench.mock.package;
```
Comment (jomarko) [inline kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/resources/org/kie/workbench/common/services/refactoring/backend/server/query/findresources/RULE4.drl]: It seems to me that this asset is not needed. Am I missing something?

File: kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/IndexingTest.java [hunk 5]
```diff
@@ -181,6 +179,35 @@ protected IOService ioService() {
             indexer.setResourceTypeDefinition(getResourceTypeDefinition());
         }
         return ioService;
+//
+//
+//        if (ioService == null) {
```
Comment (jhrcek) [inline kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/java/org/kie/workbench/common/services/refactoring/backend/server/IndexingTest.java]: Why keep this commented-out code?

File: kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/resources/org/kie/workbench/common/services/refactoring/backend/server/query/findresources/DRL4.drl (hunk 3 above)
Comment (manstis) [inline kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/resources/org/kie/workbench/common/services/refactoring/backend/server/query/findresources/DRL4.drl]: It is used [here](https://github.com/kiegroup/kie-wb-common/pull/866/files#diff-6b9760d0b5cdf14eb0c3e4c7d1f87806R165).

File: kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/resources/org/kie/workbench/common/services/refactoring/backend/server/query/findresources/RULE4.drl (hunk 4 above)
Comment (manstis) [inline kie-wb-common-services/kie-wb-common-refactoring/kie-wb-common-refactoring-backend/src/test/resources/org/kie/workbench/common/services/refactoring/backend/server/query/findresources/RULE4.drl]: It is used [here](https://github.com/kiegroup/kie-wb-common/pull/866/files#diff-6b9760d0b5cdf14eb0c3e4c7d1f87806R163).

**Comment by manstis** [comment]
@jhrcek @jomarko Updated... thanks.

**Comment by jomarko** [comment]
@manstis same here, thanks for fast reaction.

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by manstis** [comment]
Jenkins please retest this.

**Comment by manstis** [comment]
@jhrcek @jomarko This should be OK now (finally!)

**Comment by ederign** [comment]
@manstis waiting for the green here.

**Comment by ederign** [comment]
@manstis Test Result (1 failure / +1)
org.kie.workbench.common.screens.impl.LibraryServiceImplTest.queryWithAFilter

Probably is related.

**Comment by manstis** [comment]
@ederign Hi, yes possibly (probably).. looking into at the moment..

**Comment by manstis** [comment]
@ederign All green now :-)

**Pull request merged**